paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +441,88 @@ def sync_secrets(
     return True
 
 
+def sync_datastore_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+    overwrite_namespace: Optional[str] = None,
+) -> bool:
+    """
+    Map all the passwords requested for this service-instance to a single Kubernetes Secret store.
+    Volume mounts will then map the associated secrets to their associated mount paths.
+    """
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    system_paasta_config = load_system_paasta_config()
+    datastore_credentials_vault_overrides = (
```

Comment:
i was thinking that we'd add a check for
```
if not datastore_credentials_vault_overrides:
    raise Exception("Datastore credentials vault configuration is missing - refusing to continue.")
```
over checking in the secret provider - but I'll try to get my vault-tools refactor out and refactor this later :)

Reply:
maybe it should return false instead? odd how the sync scripts all return true and only true

I worry saying "overrides not existing is an error" - at which case, was it really an override or a requirement? without these overrides, it basically communicates with the default vault, which would have been the ideal situation but there are hard requirements on our vault infra for data storage

given this project is open source, maybe having it return False makes more sense?

Reply:
ah true, this would leak our internal constraints - we can definitely leave the code as-is tho :)

i think ideally we'd refactor the sync scripts a bit to make them not (as you've pointed out) unconditionally return true + handle errors in a more consistent way - but that's more work than you'd probably want to do now :p

Reply:
yep for sure, it would be ideal to refactor all of them. I'm always conflicted to fix up code smells in a feature branch. might make more sense to refactor them all as a group in a separate PR